### PR TITLE
Stacked area chart incorrect circle positions when > 10 areas

### PR DESCRIPTION
**Pull request by flacoman91**
## Description
fixing the sort algorithm to correctly transform the circle on hover over.

## Motivation and Context
Fixes this issue
https://github.com/eventbrite/britecharts/issues/682

## How Has This Been Tested?
Created sandbox and sample data and updated the code.

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/charts/stacked-area.js [hunk 1]
```diff
@@ -1019,11 +1019,14 @@ define(function(require){
...
-                .filter(v => !!v)
-                .sort((a,b) => order.indexOf(a.name) > order.indexOf(b.name));
+            let sortedValues = [];
+            values = values.filter(v => !!v);
 
-            values.forEach((d, index) => {
+            for(let i = 0; i< order.length; i++){
+                sortedValues.push(values.find(o=> o.name === order[i]));
+            }
+
```
Comment (Golodhros) [inline src/charts/stacked-area.js]: I would do something like:
```
let sortedValues = order.reduce((acc, current) => { 
    return [...acc, values.find(({name}) => name === current))]
},[]);
```
Or similar, I haven't tested it.

Comment (flacoman91) [reply src/charts/stacked-area.js]: any idea why the js sort does that with more than 10 areas?

I made a separate codepen to only demonstrate the odd behavior of the sort.
https://codepen.io/flacoman91/pen/gBZKrz?editors=0010

File: src/charts/stacked-area.js [hunk 2]
```diff
@@ -1019,11 +1019,11 @@ define(function(require){
             cleanDataPointHighlights();
 
             // ensure order stays constant
-            values = values
```
Comment (Golodhros) [inline src/charts/stacked-area.js]: You still need the values here, right?

File: src/charts/stacked-area.js (hunk 1 above)
Comment (Golodhros) [reply src/charts/stacked-area.js]: Interesting! No idea though...

File: src/charts/stacked-area.js (hunk 2 above)
Comment (flacoman91) [reply src/charts/stacked-area.js]: doesn't seem like it. It works without it.

Comment (Golodhros) [reply src/charts/stacked-area.js]: OK then

**Comment by Golodhros** [comment]
Thanks for this fix @flacoman91 !!

**Pull request merged**